scripts/line_length.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import argparse
+from typing import Optional
+from typing import Sequence
+from subprocess import call
+
+
+def main(argv: Optional[Sequence[str]] = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("filenames", nargs="*", help="Filenames to check")
+    args = parser.parse_args(argv)
+
+    for filename in args.filenames:
+        with open(filename) as fd:
+            for lineno, line in enumerate(fd):
+                if len(line) > 100:
+                    continue
+                humanlineno = lineno + 1
+                print(f"{filename}:{humanlineno} Line is too long.")
+                call(
+                    [
+                        "clang-format",
+                        "-i",
+                        f"-lines={humanlineno}:{humanlineno}",
+                        "-style={"
+                        "BasedOnStyle: Google, "
+                        "IndentWidth: 4, "
+                        "TabWidth: 8, "
+                        "UseTab: Never, "
+                        "ColumnLimit: 80, "
```

Comment:
but this is 80 here?

Reply:
This is intended.
Use 80 to 100 a allowed but not desired region.
Once you break anyway, because you exceed 100 than do it before 80.

Reply:
ok.  Can you add a comment to explain this, like:

"We only force wrapping on lines > 100, but we do so by wrapping to 80 characters for maximum compatibility with github's diff view"

Reply:
"We recommend a maximum line length of 80, but do allow up to 100 characters if deemed necessary by the developer. Lines that exceed that limit will be wrapped after 80 characters automatically.